redis/src/cluster_async/mod.rs

```diff
@@ -94,17 +94,21 @@ where
 type ConnectionFuture<C> = future::Shared<BoxFuture<'static, C>>;
 type ConnectionMap<C> = HashMap<String, ConnectionFuture<C>>;
 
-struct ClusterConnInner<C> {
-    connections: ConnectionMap<C>,
+struct ConnectionsAndMetadata<C> {
+    connections: Mutex<ConnectionMap<C>>,
     slots: SlotMap,
+    cluster_params: ClusterParams,
+}
+
+struct ClusterConnInner<C> {
+    connections_and_metadata: Arc<ConnectionsAndMetadata<C>>,
```

Comment:
Or maybe just calling this field `inner` would be sufficient and not too confusing? Anything to cut down on the width here would be spectacular.

Reply:
inner is fine with me. once we'll decide what to call the type, I'll push a commit.

Reply:
done.